vllm/v1/worker/gpu_worker.py

```diff
@@ -359,7 +359,16 @@ def execute_model(
                 get_pp_group().recv_tensor_dict(
                     all_gather_group=get_tp_group()))
 
-        output = self.model_runner.execute_model(scheduler_output,
+        # add trace annotation so that we can easily distinguish new/cached request numbers in each iteration
+        context = nullcontext()
+        if self.profiler:
+            num_new_reqs = len(scheduler_output.scheduled_new_reqs)
+            num_cached_reqs = len(scheduler_output.scheduled_cached_reqs.req_ids)
+            context = torch.profiler.record_function(
+                f"execute_{num_new_reqs}_{num_cached_reqs}")
```

Comment:
1. shouldn't this use torch nvtx ranges? 
2. `execute_new_{num_new...}_cached_{num_cached...}`
3. also this doesn't really help explaining prefill vs decode because there's prefix caching involved. can you help think about more useful information to annotation?